### PR TITLE
fix(client): allow read to contain empty fields

### LINKED ISSUE #22: OpenFgaClient - Read sends an empty user/relation/object when not provided

### PR DESCRIPTION
**Pull request by rhamzeh**
## References

Closes #22 

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -885,15 +885,18 @@ func (request SdkClientReadRequest) Execute() (*ClientReadResponse, error) {
 }
 
 func (client *OpenFgaClient) ReadExecute(request SdkClientReadRequest) (*ClientReadResponse, error) {
-	data, _, err := client.OpenFgaApi.Read(request.ctx).Body(openfga.ReadRequest{
-		TupleKey: &openfga.TupleKey{
+	body := openfga.ReadRequest{
+		PageSize:          getPageSizeFromRequest((*ClientPaginationOptions)(request.options)),
+		ContinuationToken: getContinuationTokenFromRequest((*ClientPaginationOptions)(request.options)),
+	}
+	if request.body != nil && request.body.User != nil || request.body.Relation != nil || request.body.Object != nil {
```
Comment (adriantam) [inline client/client.go]: Do we want to raise error in the false case or we rely on the API to do that?

Comment (rhamzeh) [reply client/client.go]: It shouldn't error, we accept sending just a nil user, relation or object

Comment (rhamzeh) [reply client/client.go]: The problem is that when none are sent, we used to send an empty tuple key which caused the API to through an error

Comment (adriantam) [inline client/client.go]:
```suggestion
	if request.body != nil && (request.body.User != nil || request.body.Relation != nil || request.body.Object != nil) {
```

**Review by adriantam — CHANGES_REQUESTED** [review]

**Review by adriantam — APPROVED** [review]

**Pull request merged**